a3/Project Scheduling.md

```diff
@@ -0,0 +1,212 @@
+
+The developers are: Andrii, Josh, Ryan, and Khush.
+
+The tasks assigned to Andrii are:
+- 4.1.1 Save Contacts
+- 4.1.2 Delete Contacts
+- 4.1.3 Edit Contacts
+- 4.2.1 Interface Layout
+- 2.3 Blacklist Unpaying Users
+- 1.1.2 Grab Logs from Backend
+
+The tasks assigned to Josh are:
+- 4.1.4 Call Another User
+- 4.2.2 Interface Style
+- 4.2.4.1 Dial Unsaved Number
+- 4.2.4.2 Dial Saved Number
+- 3.2 Call Routing
+- 1.1.2 Look at User Bills
+
+
+The tasks assigned to Ryan are:
+- 4.1.5 Accept Incoming Calls
+- 4.1.6 Decline Incoming Calls
+- 4.2.4.3 Decline Call
+- 3.3 Track Call Logs
+- 2.2 Inform Users
+- 1.2.1 Moderate Users
+
+The tasks assigned to Khush are:
+- 3.1 User Database
+- 4.2.3.1 Incoming Call
+- 4.2.3.2 Missed Call
+- 4.2.3.3 Busy Call
+- 4.2.3.4 Declined Call
+- 2.1 Calculate User Bills
+- 1.1.1 Look at Users Logs
+
+# 2.3 Time Estimates
+
+All time measurements are measured in days, assuming a full 8 hour work day.
+
+All time estimates made through consulting an expert.
+
+| Task Number | Task Name | Optimistic | Pessimistic | Most Likely |
+| ----------- | --------- | ---------- | ----------- | ----------- |
+| 1.1.1 | Look at User Logs | 1 | 3 | 2 |
+| 1.1.2 | Look at User Bills | 1 | 3 | 2 |
+| 1.2.1 | Moderate Users | 4 | 10 | 6 |
+| 1.2.2 | Grab Logs from Backend | 1 | 2 | 1 |
+| 2.1 | Calculate User Bills | 1 | 3 | 2 |
+| 2.2 | Inform Users | 3 | 5 | 4 |
+| 2.3 | Blacklist Unpaying Users | 1 | 7 | 4 |
+| 3.1 | User Database | 3 | 10 | 5 |
+| 3.2 | Call Routing | 8 | 20 | 14 |
+| 3.3 | Track Call Logs | 3 | 6 | 4 |
+| 4.1.1 | Save Contacts | 1 | 4 | 2 |
+| 4.1.2 | Delete Contacts | 1 | 4 | 2 | 
+| 4.1.3 | Edit Contacts | 1 | 4 | 2 |
+| 4.1.4 | Call Another User | 4 | 8 | 6 |
+| 4.1.5 | Accept Incoming Calls | 2 | 4 | 3 |
+| 4.1.6 | Decline Incoming Calls | 1 | 2 | 2 |
+| 4.2.1 | Interface Layout | 10 | 15 | 12 |
+| 4.2.2 | Interface Styling | 10 | 15 | 12 |
+| 4.2.3.1 | Incoming Call | 1 | 3 | 2 |
+| 4.2.3.2 | Missed Call | 1 | 3 | 2 |
+| 4.2.3.3 | Busy Call | 1 | 3 | 2 |
+| 4.2.3.4 | Declined Call | 2 | 4 | 3 |
+| 4.2.4.1 | Dial Unsaved Number | 2 | 5 | 3 |
+| 4.2.4.2 | Dial Saved Number | 2 | 5 | 3 |
+| 4.2.4.3 | Decline Call | 2 | 7 | 5 |
+
+# 2.4 Task Order
+
+#### Independent Tasks:
+
+These tasks can be started at any time, though there may be tasks depending on them.
+
+- 3.1
+- 4.2.1
+- 4.2.2
+
+#### Finish-to-Start Dependencies:
+
+These tasks depend on the previous task finishing before they can start.
+
+Denoted:
+A -> B 
+
+Which defines: The successor task 'B' requires the predecessor task 'A' to finish before it can start
+
+| Predecessor | Successor |
+| ----------- | --------- |
+|3.1|3.2|
+|3.1|3.3|
+|3.1|4.1.1|
+|3.1|4.1.2|
+|3.1|4.1.3|
+|3.2|4.1.4|
+|3.2|4.1.5|
+|3.2|4.1.6|
+|3.2|4.2.3.1|
+|3.2|4.2.3.2|
+|3.2|4.2.3.3|
+|3.2|4.2.3.4|
+|3.2|4.2.4.1|
+|3.2|4.2.4.2|
+|3.2|4.2.4.3|
+|3.3|2.1|
+|3.3|1.1.1|
+|3.3|1.2.2|
+|2.1|2.2|
+|2.1|2.3|
+|2.1|1.1.2|
+|2.3|1.2.1|
```

Comment:
In order to blacklist unpaying users (2.3), you first need to moderate users (1.2.1) to identify those who are not paying and should be blacklisted. Therefore, 1.2.1 is a prerequisite for 2.3, and the dependency should flow from 1.2.1 to 2.3.